### PR TITLE
Auth devise

### PR DESCRIPTION
**Pull request by sajibarafatsiddiqui**
- Removed current_user method in ApplicationController, devise will provide us with one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
Following features added:
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed password should be stored in the database.
- Ask for confirmation of email.
- Can reset password.

### CODE REVIEW

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @sajibarafatsiddiqui  👋 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ⚠️ 
- [ ] Please add an approval screenshot from student success to your pull request description to show proof that you have been allowed to work alone on this pair programming project. This is a requirement from the guidelines. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by sajibarafatsiddiqui** [comment]
image

**Review by Gambit142 — COMMENTED** [review]
Hi @sajibarafatsiddiqui 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- I can see that you attached a screenshot of correspondence from student success, but we can't see who sent the correspondence and the date that the correspondence was sent. Please attach a screenshot of the full correspondence. Thank you. 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by sajibarafatsiddiqui** [comment]
@Gambit142 R u that responsible to invalidate this request without asking me in slack or it is easy to invalidate a request before judging than take time in reviewing code?

image

**Comment by sajibarafatsiddiqui** [comment]
image

**Comment by sajibarafatsiddiqui** [comment]
> Hi @sajibarafatsiddiqui 👋,
> 
> While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
> 
> ## Invalid Code Review Request
> * I can see that you attached a screenshot of correspondence from student success, but we can't see who sent the correspondence and the date that the correspondence was sent. Please attach a screenshot of the full correspondence. Thank you. 👍
> 
> Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
> 
> Cheers, and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _Invalid Code Review Request **does not count** into the code reviews limit._

I am astonished!!!

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (ridwanediallo) [inline app/views/devise/sessions/new.html.erb]: - [ ] Please ensure that the form for login a user matches the one provided in the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_login.png), as stated in the requirement.

#### Requirement :point_down: 
Screenshot from 2023-05-19 19-28-14

### Your form so far :point_down: 
Screenshot from 2023-05-19 19-18-25

### Wireframe :point_down: 
Screenshot from 2023-05-25 14-52-14

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (ridwanediallo) [inline app/views/devise/registrations/new.html.erb]: - [ ] Please ensure that the form for registering a user matches the one provided in the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_login.png), as stated in the requirement.

#### Requirement :point_down: 
Screenshot from 2023-05-19 19-28-14

### Your form so far :point_down: 
Screenshot from 2023-05-19 19-18-43

### Wireframe :point_down: 
Screenshot from 2023-05-25 14-52-14

**Review by ridwanediallo — CHANGES_REQUESTED** [review]
# Changes Required ♻️
## Hi @sajibarafatsiddiqui , 
AlmostThereReachGIF

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Your readme file looks professional. ✔


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/new.html.erb (hunk 2 above)
Comment (sajibarafatsiddiqui) [reply app/views/devise/registrations/new.html.erb]: @ridwanediallo how I will register a user by login form? Please suggest me

Comment (ridwanediallo) [reply app/views/devise/registrations/new.html.erb:32]: @sajibarafatsiddiqui are you available on Zoom?

Comment (sajibarafatsiddiqui) [reply app/views/devise/registrations/new.html.erb]: change it accordingly
image
image

**Review by TadesseDev — APPROVED** [review]
### Hi @sajibarafatsiddiqui 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅
## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**